File: tests/samples/vulnerable_sample.java
import java.sql.*;
import java.io.*;

public class VulnerableSample {

    // Vulnerability 1: Hardcoded credentials (Semgrep: java.lang.security)
    private static final String PASSWORD = "admin123";
    private static final String DB_URL = "jdbc:mysql://localhost/db?user=root&password=root123";

    // Vulnerability 2: SQL Injection — string concatenation in query
    public static void getUser(String username) throws Exception {
        Connection conn = DriverManager.getConnection(DB_URL);
        Statement stmt = conn.createStatement();
        String query = "SELECT * FROM users WHERE username = '" + username + "'";
        ResultSet rs = stmt.executeQuery(query);
    }

    // Vulnerability 3: Command injection — Runtime.exec with user input
    public static void runCommand(String userInput) throws Exception {
        Runtime runtime = Runtime.getRuntime();
        runtime.exec("ls " + userInput);
    }

    // Clean method — should NOT be flagged
    public static int safeAdd(int a, int b) {
        return a + b;
    }

    // Vulnerability 4: Weak hashing algorithm (MD5)
    public static void hashPassword(String password) throws Exception {
        java.security.MessageDigest md = java.security.MessageDigest.getInstance("MD5");
        byte[] hash = md.digest(password.getBytes());
    }
}

File: tests/samples/vulnerable_sample.py
import sqlite3
import subprocess

# Vulnerability 1: Hardcoded password (Bandit: B105)
PASSWORD = "admin123"
DB_PASSWORD = "supersecret"

def get_user(username):
    # Vulnerability 2: SQL Injection — user input directly in query (Bandit: B608, Semgrep)
    conn = sqlite3.connect("users.db")
    cursor = conn.cursor()
    query = "SELECT * FROM users WHERE username = '" + username + "'"
    cursor.execute(query)
    return cursor.fetchall()

def run_command(user_input):
    # Vulnerability 3: Shell injection — user input passed to shell (Bandit: B602)
    subprocess.call(user_input, shell=True)

def safe_add(a, b):
    # This is clean — should NOT be flagged
    return a + b

def calculate(expression):
    # Vulnerability 4: eval() on user input (Bandit: B307)
    return eval(expression)